### PR TITLE
fetching popular bookmarks based on clicks

### PR DESCRIPTION
**Pull request by kshitijiiit**
- Fixed API for fetching popular bookmarks.
- Added views for popular bookmarks.
- Option for viewing popular bookmarks (both personal and all) in the footer
- Added API tests

### CODE REVIEW

File: bookie/views/api.py [hunk 1]
```diff
@@ -393,8 +400,6 @@ def user_bmark_count(request):
     })
 
 
-@view_config(route_name="api_bmarks_popular", renderer="jsonp")
-@view_config(route_name="api_bmarks_popular_user", renderer="jsonp")
 @api_auth('api_key', UserMgr.get, anon=True)
```
Comment (mitechie) [inline bookie/views/api.py]: we should just remove this method then since it's no longer part of the api.

File: bookie/views/api.py [hunk 2]
```diff
@@ -306,6 +306,13 @@ def bmark_recent(request, with_content=False):
     page = int(params.get('page', '0'))
     count = int(params.get('count', RESULTS_MAX))
 
+    # check if we have a ordering by popularity specified
```
Comment (mitechie) [inline bookie/views/api.py]: please update this to be a complete sentence with a capital letter and ending in a period.

File: bookie/static/js/bookie/api.js [hunk 3]
```diff
@@ -329,6 +329,7 @@ YUI.add('bookie-api', function (Y) {
             data: {
                 count: 10,
                 page: 0,
+                sort: 'popular',
```
Comment (mitechie) [inline bookie/static/js/bookie/api.js]: we don't want to default to sorting by popular

File: bookie/views/api.py [hunk 4]
```diff
@@ -306,6 +306,13 @@ def bmark_recent(request, with_content=False):
...
+    order_by = params.get('sort', None)
+    if order_by == "popular":
+        order_by = Bmark.clicks.desc()
```
Comment (mitechie) [inline bookie/views/api.py:325]: Do we have click values? Should this instead be Bmark.hashed.click.desc()?

Remember we've got two click counts. one per user and one for the url as a whole?

**Comment by mitechie** [comment]
So the change looks ok on the surface. I'd check the database and make sure you have real click values to sort on.

The other thing you can do to help debug is to make manual api requests and see if you get different results against your test system.

I use a tool called httpie, which is written in python and provides a pretty output. https://pypi.python.org/pypi/httpie/0.8.0

You can then use that to manually build a request to the api and add the flag and see how your results change or not.

File: bookie/views/api.py (hunk 1 above)
Comment (kshitijiiit) [inline bookie/views/api.py]: Yes, I'll do that in the next commit.

File: bookie/views/api.py (hunk 2 above)
Comment (kshitijiiit) [inline bookie/views/api.py]: Sure.

File: bookie/static/js/bookie/api.js (hunk 3 above)
Comment (kshitijiiit) [inline bookie/static/js/bookie/api.js]: Yes, I know but I was just trying this out to see if it works. And it doesn't seem to reflect the changes in the final view. Would you know the reason for it? 
The api seems to be working absolutely fine but I really can't find out how to reflect that change in the view of "/recent".

File: bookie/views/api.py (hunk 4 above)
Comment (kshitijiiit) [inline bookie/views/api.py:325]: Bmark.hashed.click is the global click count for a given url. I think we would need to sort by local clicks for each user in this api which is given by Bmark.clicks. Perhaps a different call can be made to sort by the global count. What do you think?

**Comment by mitechie** [comment]
Thanks so much for this change. I've landed it and will update it on the live site soon.

**Pull request closed without merging**